### ISSUE TITLE
Add Auto Save to YAML Storage

### ISSUE THREAD

**Issue by Mirayii** [Auxilor/eco#219]
**Is your feature request related to a problem? Please describe.**
We noticed that after a server crash, when the plugin isn't forced to save player data, the plugin isn't saving them to the YAML file.

**Describe the solution you'd like**
Add an Auto Save feature with interval (like save once per 10 minutes) for YAML Storage type.

**Describe alternatives you've considered**
We tried to switch to MySQL, but we have found it to be more laggy and sometimes losing data (not saving them again frequently? save-interval was on 20 ticks)

**Additional context**
We have tried to look up the problem ourselves, but it looks like the data rollback only happens when the server crashes. After Restart, it works as intended.

**Comment by SignorPollito** [comment]
Yeah, all the "Eco Suite", when get player data, they make a request to the database every time, so on populated servers it may get laggy.

They should implement a cache system so that the plugin doesn't have to get the data every time from the database or the file.
And when the player quits, they just have to save the data and remove it from the cache.

There is also another problem: if a player switches a server very fast, the database will not have the time to save and the other server will get the older data instead of the new saved one.
